### PR TITLE
Generating a .deb for status-desktop

### PR DESCRIPTION
**Pull request by richard-ramos**
Requires
- https://github.com/status-im/status-go/pull/2370

### Instructions
1. Create the docker image 
```
cd deb
docker build -t statusteam/status-desktop-deb-build:latest .
cd ..
```
2. Execute `./build-linux-deb.sh`

**TODO**
- [x] Create a dockerfile with the required packages for building the .deb (Starting from version 21.04 Hirsute Hippo, Ubuntu and Debian bullseye include QT 5.15.2)
- [x] Make sure the dockerfile does not use a manually installed QT. It should use the distro libs
- [x] Edit control file and add version of the dependencies (see virtualbox .deb, their QT requirements are similar)
- [x] Add status-desktop to gnome menu
- [x] Add gstreamer dependencies to control file

In a future PR:
- Add dictionaries to .deb (Depends on https://github.com/status-im/status-desktop/issues/3448)

### CODE REVIEW

**Comment by richard-ramos** [comment]
I had to execute
```
unset QTDIR
 unset QT_PATH
```
so it uses the OS QT libs on /usr/lib/x86_64-linux-gnu instead of those of the manual installation of QT

**Comment by richard-ramos** [comment]
The following packages are required in Ubuntu 21.04:
```
pbuilder ubuntu-dev-tools apt-file qtbase5-dev qtdeclarative5-dev qml-module-qt-labs-platform qtquickcontrols2-5-dev libqt5svg5-dev libqt5gui5 libqt5core5a libqt5network5 libqt5qml5 libqt5quick5 libqt5svg5 libqt5webengine5 libqt5webenginecore5 libqt5quickcontrols2-5 libqt5widgets5 
```

**Comment by richard-ramos** [comment]
Installing the resulting .deb and executing `status-desktop` displays the following error:
```
status@laptop:~/status-desktop$ status-desktop
/home/status/status-desktop/src/nim_status_client.nim(269) nim_status_client
/home/status/status-desktop/src/nim_status_client.nim(45) mainProc
/home/status/status-desktop/vendor/nimbus-build-system/vendor/Nim/lib/system/io.nim(693) readFile
Error: unhandled exception: cannot open: /usr/local/fleets.json [IOError]
```
**TODO**: make the content of fleets.json part of a defined constant in nim, instead of reading it from the .json

**Comment by richard-ramos** [comment]
The new dockerfile uses debian:bullseye instead of ubuntu:21.04,  so it uses a slightly older version of libc, and also because in this version Qt 5.15.2 is introduced. The resulting .deb works in Debian, Ubuntu and Pop OS.

File: build-linux-appimage.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# This script assumes $PWD is the same dir in which this script is located
+
+# Helps avoid permissions problems with `jenkins` user in docker container when
+# making a local packaged build
+git clean -dfx
+
+# Build .AppImage
+docker run -it --rm \
```
Comment (jakubgs) [inline build-linux-appimage.sh:10]: Could we maybe move these scripts into `scripts`? The root of this repos is already cluttered enough as it is.

File: Makefile [hunk 2]
```diff
@@ -453,6 +493,8 @@ pkg-windows: check-pkg-target-windows $(STATUS_CLIENT_EXE)
 
 zip-windows: check-pkg-target-windows $(STATUS_CLIENT_7Z)
 
+deb: check-pkg-target-linux $(STATUS_CLIENT_DEB)
```
Comment (jakubgs) [inline Makefile:496]: Shouldn't this be `deb-linux`? Or really, the targets should start with the platform, so it's:
```
make linux-deb
make linux-app
make windows-zip
make windows-exe
```

File: scripts/deb/control [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Package: status-desktop
```
Comment (jakubgs) [inline scripts/deb/control:1]: None of these files are scripts, and should not be under `scripts`. As stated in #3367 we need something like a `resources` folder where we can put stuff like that. So it's `resources/pkg/deb/control` or something like that.

File: docker-linux.sh [hunk 4]
```diff
@@ -5,8 +5,7 @@ cp -R . ~/status-desktop
 cd ~/status-desktop
 
 git clean -dfx && rm -rf vendor/* && make -j4 V=1 update
-make V=1 pkg
+make V=1 $1
```
Comment (jakubgs) [inline docker-linux.sh:8]: This should also be in `scripts`, not in repo root.

File: ci/README.md [hunk 5]
```diff
@@ -9,9 +9,17 @@ https://ci.status.im/job/nim-status-client/
 
 In order to build the Linux version of the application we use a modified `a12e/docker-qt:5.14-gcc_64` Docker image with the addition of Git and Golang.
 
-The image is built with [`Dockerfile`](./Dockerfile) using:
+The image is built with Docker using:
 ```
+# for building the AppImage
+cd AppImage
 docker build -t statusteam/nim-status-client-build:latest .
+cd ..
```
Comment (jakubgs) [inline ci/README.md:17]: Don't you mean `ci/AppImage`? Also, you can do this like this:
```
docker build -t statusteam/nim-status-client-build:latest ci/AppImage
```
Without having to `cd` anywhere`.

**Review by jakubgs — CHANGES_REQUESTED** [review]
Looks decent, but the location of some of the files doesn't make sense.

**Review by jakubgs — APPROVED** [review]
Much better.

**Pull request merged**